### PR TITLE
Refactor Font configuration UI and resources.

### PR DESCRIPTION
**Pull request by bruvzg**
Ensures all font properties are accessible during runtime, as a single resource.

- Merges all import time properties of `Font` and `FontData` into single `Font` resource. Including fallback font list and pre-rendered glyphs (different sets of pre-rendered glyphs can be selected for different sizes/variations).
- Merges all runtime properties: `font_size`, OpenType variations (was part of `Font`), embolden/slant (was not exposed), OpenType features (was control property), and extra spacing (was partially accessible by code) into single `FontConfig` resources.

Fully collapsed `FontConfig` resource:
Screenshot 2022-05-25 at 12 32 30

Fully expanded `FontConfig` resource, with all advanced settings:
Screenshot 2022-05-25 at 12 33 14

- Adds a new inspector plugin for the `FontConfig` (for editing  OpenType variation and feature Dictionaries), replaces old OpenType features plugin.
- Removes font config toolbar.
- Font selecting menu is moved from the toolbar to the `Font` resource dropdown:
Screenshot 2022-05-25 at 12 33 22

TODO: 

- [x] Add convenient way to access `FontConfig` for the "text controls", as a separate property or `InspectorPlugin` to display relevant theme overrides directly under the `text` property?
- [x] Test all `change` callback and fallback selection is working properly.
- [x] Provide access to all `FontConfig` properties via RTL `[font ..]` bbcode tag.
- [x] Update documentation.

### CODE REVIEW

**Comment by bruvzg** [comment]
Some other loosely related properties that might need some attention, but probably as a separate PR:
- "BiDi Direction" and "BiDi override" properties: used only in some rare edge cases, 99% of user won't need to change default values, so it’s probably OK to keep them in the collapsed property group at the bottom of the list.
- Outlines, label shadow: it's highly inconsistent right now, and inconvenient to use (size and color are different type). Might be worth adding a new `Outlines` type of theme properties (as an Array of a new `TextOutline` (color, sized, offset) objects) to replace all of them, also will provide option to have multiple stacked outlines and shadows.

File: scene/resources/font.cpp [hunk 1]
```diff
@@ -1526,798 +1522,846 @@ void FontData::remove_cache(int p_cache_index) {
...
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "size", PROPERTY_HINT_RANGE, "1,4000,1"), "set_size", "get_size");
 
-	ClassDB::bind_method(D_METHOD("get_rids"), &Font::get_rids);
+	ADD_GROUP("Variation", "variation");
+	ADD_PROPERTY(PropertyInfo(Variant::DICTIONARY, "variation_opentype"), "set_variation_opentype", "get_variation_opentype");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "variation_embolden", PROPERTY_HINT_RANGE, "-2,2,0.01"), "set_variation_embolden", "get_variation_embolden");
+	ADD_PROPERTY(PropertyInfo(Variant::TRANSFORM2D, "variation_transform"), "set_variation_transform", "get_variation_transform");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "variation_customize_fallbacks"), "set_variation_customize_fallbacks", "get_variation_customize_fallbacks");
+
+	ADD_GROUP("OpenType features", "opentype");
```
Comment (fire-forge) [inline scene/resources/font.cpp:1964]:
```suggestion
	ADD_GROUP("OpenType Features", "opentype");
```
Just something small I noticed when looking at the screenshots.

**Comment by bruvzg** [comment]
For selected controls, `Font` is now available right under the `text` (it's just a convenience shortcut for the theme override), also moved all BiDi properties to the bottom of the list.

Screenshot 2022-05-26 at 10 15 44

**Comment by YuriSizov** [comment]
I like the order that this brings, but this has a huge downgrade in usability, as changing font size, something that is done __very__ often once again requires duplication and modification of resources. Maybe with SDF fonts this won't be required as much in 4.0, and fonts would be crisp and scalable, but I have no experience with them to be sure.

**Comment by Zireael07** [comment]
>  like the order that this brings, but this has a huge downgrade in usability — as changing font size, something that is done very often, once again requires duplication and modification of resources. Maybe with SDF fonts this won't be required as much in 4.0, and fonts would be crisp and scalable, but I have no experience with them to be sure.

Changing font size shouldn't require duplicating resources IMHO, I often find myself using the same font in different sizes in different places

**Comment by YuriSizov** [comment]
> > like the order that this brings, but this has a huge downgrade in usability — as changing font size, something that is done very often, once again requires duplication and modification of resources. Maybe with SDF fonts this won't be required as much in 4.0, and fonts would be crisp and scalable, but I have no experience with them to be sure.
> 
> Changing font size shouldn't require duplicating resources IMHO, I often find myself using the same font in different sizes in different places

Yes, that's my point as well. Although in this case your font resource will be separate from your font config anyway, so you don't have to duplicate the font itself, only its config. So it's mainly a usability issue to me, as changing a size as a number vs duplicating and modifying a resource with a size inside is not equally intuitive.

**Comment by bruvzg** [comment]
> Although in this case your font resource will be separate from your font config anyway.

Unlike old 3.x fonts, `FontConfig` duplication won't cause actual font duplication, only a tiny config class. But I guess I'll make an alternative version with font size left intact. There's noting that requires it to be in a `FontConfig`, I just prefer to have everything in a one place.

**Comment by YuriSizov** [comment]
> I just prefer to have everything in a one place.

This is indeed appealing. Well, since 4.1 would probably still feature breaking changes, we can try it this way and re-introduce dedicated size property for 4.1 if this doesn't work well for users.

**Comment by bruvzg** [comment]
> But I guess I'll make an alternative version with font size left intact.

Here it is: #61473, the same changes to import, but `font_size` is a separate theme property, instead of part of `FontConfig`.

**Comment by bruvzg** [comment]
Superseded by #61473 (same but with a `font_size` as a separate theme property).

**Pull request closed without merging**